tests/tt_metal/tt_metal/unit_tests_fast_dispatch/command_queue/test_EnqueueTrace.cpp

```diff
@@ -145,24 +148,26 @@ TEST_F(CommandQueueFixture, EnqueueProgramTraceCapture) {
 
     EnqueueWriteBuffer(command_queue, input, input_data.data(), true);
 
-    uint32_t tid = BeginTraceCapture(this->device_, command_queue.id(), 2048);
+    uint32_t tid = BeginTraceCapture(device, command_queue.id());
     EnqueueProgram(command_queue, simple_program, false);
-    EndTraceCapture(this->device_, command_queue.id(), tid);
+    EndTraceCapture(device, command_queue.id(), tid);
 
     EnqueueTrace(command_queue, tid, true);
     EnqueueReadBuffer(command_queue, output, trace_output_data.data(), true);
     EXPECT_TRUE(eager_output_data == trace_output_data);
 
     // Done
     Finish(command_queue);
-    ReleaseTrace(this->device_, tid);
+    ReleaseTrace(device, tid);
+    tt::tt_metal::CloseDevice(device);
 }
 
-TEST_F(CommandQueueFixture, EnqueueProgramDeviceCapture) {
-    Buffer input(this->device_, 2048, 2048, BufferType::DRAM);
-    Buffer output(this->device_, 2048, 2048, BufferType::DRAM);
+TEST(TraceTests, EnqueueProgramDeviceCapture) {
+    Device* device = tt::tt_metal::CreateDevice(0, 1, 0, 2048);
```

Comment:
is the magic 2048 the size?  does it have to match the Buffer sizes below?  use a named constexpr?

Reply:
2048 is the size of the Trace buffer (user needs to know beforehand). Those other buffers store activations. There isn't really a relation between these sizes.